data_structures/graphs/prim.c

```diff
@@ -0,0 +1,229 @@
+#include<stdio.h>
+#include<stdlib.h>
+#define max_edges_of_vertex 100000
+/*      max_edges_of_vertex will be used to initialize adjacency list
+        you can modify size according to question constraint                        */
+
+#define INFINITY LONG_LONG_MAX
+
+// a structure to represent a weighted edge in graph
+typedef struct NODE
+{
+	long long u;
+	long long w;
+	struct NODE* next;
```

Comment:
Please add a one-line description of what the variable is for (**see the example below**).

```c
int64_t data = 0;       ///< The value/key of the node.
int64_t array[64];      ///< The array that will be used for the tree.
```